quantizers/uniform.py

```diff
@@ -59,9 +63,27 @@ def quantize(self, X):
             return np.zeros(X.shape) + center, total_bytes
 
         # Actually quantize.
-        compressed_X = _quantize(X_recentered, self.num_bits, sf)
+        compressed_X = self._quantize(X_recentered, self.num_bits, sf)
 
         # Recenter and return.
         compressed_X += center
 
         return compressed_X, total_bytes
+
+
+class MidtreadQuantizer(UniformQuantizer):
+    def name(self):
+        return "uniform_mt" + str(self.num_bits) + "b"
+
+    def quantize(self, X):
+        if (self.num_bits <= 1):  #mid-tread requires at least 2 bits
+            return np.zeros(X.shape), UniformQuantizer.get_total_bytes(self, X)
+
+        L = 2**self.num_bits - 1
```

Comment:
shouldn't this be 2**(self.num_bits-1)?

Reply:
For clarity please change this to "(2**(self.num_bits-1))-1" please add a comment saying this is the maximum positive number you can represent.